docs/scripts_md/MRIProcessingUtility.md

```diff
@@ -0,0 +1,379 @@
+  - $User         : user that is running the pipeline
+  - $centerID     : center ID
+
+### setMRISession($subjectIDsref, $tarchiveInfo)
+
+Sets the imaging session ID. This function will call
+`&NeuroDB::MRI::getSessionID` which in turn will either:
+  - grep sessionID if visit for that candidate already exists, or
+  - create a new session if visit label does not exist for that
+     candidate yet
+
+INPUTS: subject's ID information hashref, tarchive information hashref
+
+RETURNS: session ID, if the new session requires staging
+
+### validateArchive($tarchive, $tarchiveInfo)
+
+Validates the DICOM archive by comparing the MD5 of the `$tarchive file` and
+the one stored in the tarchive information hashref `$tarchiveInfo` derived
+from the database. The function will exits with an error message if the
+tarchive is not validated.
+
+INPUTS: tarchive file, tarchive information hashref
+
+### which\_directory($subjectIDsref, $data\_dir)
+
+Determines where the MINC files to be registered into the database will go.
+
+INPUTS: subject's ID information hashref, data directory (typically
+/data/project/data)
+
+RETURNS: the final directory in which the registered MINC files will go
+(typically /data/project/data/assembly/CandID/visit/mri/)
+
+### validateCandidate($subjectIDsref, $tarchive\_srcloc)
+
+Check that the candidate's information derived from the patient name field of
+the DICOM files is valid (CandID and PSCID of the candidate should correspond
+to the same subject in the database).
+
+INPUTS: subject's ID information hashref, tarchive source location
+
+RETURNS: the candidate mismatch error, or undef if the candidate is validated
+or a phantom
+
+### computeSNR($tarchiveID, $tarchive\_srcloc, $profile)
+
+Computes the SNR on the modalities specified in the `getSNRModalities()`
+routine of the `$profile` file.
+
+INPUTS: tarchive ID, tarchive source location, configuration file (usually prod)
+
+### orderModalitiesByAcq($tarchiveID, $tarchive\_srcloc)
+
+Order imaging modalities by acquisition number.
+
+INPUTS: tarchive ID, tarchive source location
+
+### getUploadIDUsingTarchiveSrcLoc($tarchive\_srcloc)
+
+Gets the upload ID form the `mri_upload` table using the DICOM archive
+`SourceLocation` specified in the `tarchive` table.
+
+INPUT: DICOM archive's source location
+
+RETURNS: the found upload ID
+
+### spool($message, $error, $upload\_id, $verb)
+
+Calls the Notify->spool function to log all messages.
+
+INPUTS:
+  - $message   : message to be logged in the database
+  - $error     : if 'Y' it's an error log,
+                 'N' otherwise
+  - $upload\_id : the upload\_id
+  - $verb      : 'N' for few main messages,
+                 'Y' for more messages (developers)
+
+# TO DO
+
+Document the following functions:
+  - get\_acqusitions($study\_dir, \\@acquisitions)
+  - concat\_mri($minc\_files)
+  - registerProgs(@toregister)
+
+Remove function get\_acqusitions($study\_dir, \\@acquisitions) that is not used
```

Comment:
I think you meant to keep either line 362 or line 366 :)

Reply:
@MounaSafiHarab Good catch! Should be resolved now. Thanks!